app/kuma-dp/pkg/dataplane/envoy/remote_bootstrap.go

```diff
@@ -71,15 +77,24 @@ func (b *remoteBootstrap) Generate(url string, cfg kuma_dp.Config) (proto.Messag
 	return &bootstrap, nil
 }
 
-func (b *remoteBootstrap) requestForBootstrap(url *net_url.URL, cfg kuma_dp.Config) ([]byte, error) {
+func (b *remoteBootstrap) requestForBootstrap(url *net_url.URL, cfg kuma_dp.Config, dp *core_mesh.DataplaneResource) ([]byte, error) {
 	url.Path = "/bootstrap"
+	var dataplaneResource string
+	if dp != nil {
+		dpYAML, err := yaml.Marshal(rest_types.From.Resource(dp))
+		if err != nil {
+			return nil, err
+		}
+		dataplaneResource = base64.StdEncoding.EncodeToString(dpYAML)
+	}
```

Comment:
I think it makes more sense to pass it as JSON as a string, not with YAML encoded to base64. It will be easier to debug.